applications/tests/test_reservation_creation.py

```diff
@@ -0,0 +1,245 @@
+            application_event_schedule=cls.schedule,
+            allocated_reservation_unit=cls.res_unit,
+            allocated_begin=datetime.time(hour=12),
+            allocated_end=datetime.time(hour=14),
+        )
+        cls.service_sector_manager = User.objects.create(
+            username="service_sector_person",
+            first_name="ser",
+            last_name="vice",
+            email="ser.vice.person@foo.com",
+        )
+        reservation_date = next_or_current_matching_weekday(
+            cls.application_event.begin, cls.result.allocated_day
+        )
+        cls.reservation_begin = datetime.datetime.combine(
+            reservation_date, cls.result.allocated_begin, tzinfo=get_default_timezone()
+        )
+        cls.reservation_end = datetime.datetime.combine(
+            reservation_date, cls.result.allocated_end, tzinfo=get_default_timezone()
+        )
+
+    def test_reservation_time_equal_reservation_time_when_open_in_reservation_time(
+        self, mock
+    ):
+        scheduler = ReservationScheduler(
+            self.res_unit.id, self.reservation_begin, self.reservation_end
+        )
+        start_dt, end_dt = scheduler.get_reservation_times_based_on_opening_hours()
+
+        assert_that(start_dt).is_equal_to(self.reservation_begin)
+        assert_that(end_dt).is_equal_to(self.reservation_end)
+
+    def test_reservation_time_is_one_hour_reduced_from_reservation_time_when_closes_before_reservation_end(
+        self, mock
+    ):
+        reservation_end = self.reservation_end + datetime.timedelta(hours=1)
+        scheduler = ReservationScheduler(
+            self.res_unit.id, self.reservation_begin, reservation_end
+        )
+        start_dt, end_dt = scheduler.get_reservation_times_based_on_opening_hours()
+
+        expected_res_end = datetime.datetime(
+            end_dt.year, end_dt.month, end_dt.day, 14, 0, tzinfo=get_default_timezone()
+        )
+        assert_that(start_dt).is_equal_to(self.reservation_begin)
+        assert_that(end_dt).is_equal_to(expected_res_end)
+
+    def test_reservation_time_is_one_hour_reduced_from_reservation_time_when_opens_after_reservation_begin(
+        self, mock
+    ):
+        reservation_start = self.reservation_begin - datetime.timedelta(hours=1)
+        scheduler = ReservationScheduler(
+            self.res_unit, reservation_start, self.reservation_end
+        )
+        start_dt, end_dt = scheduler.get_reservation_times_based_on_opening_hours()
+
+        expected_res_begin = datetime.datetime(
+            start_dt.year,
+            start_dt.month,
+            start_dt.day,
+            12,
+            0,
+            tzinfo=get_default_timezone(),
+        )
+        assert_that(start_dt).is_equal_to(expected_res_begin)
+        assert_that(end_dt).is_equal_to(self.reservation_end)
+
+    def test_reservation_times_are_none_when_the_reservation_unit_is_not_open_in_reservation_time(
```

Comment:
KInd of same here. I think maybe we should create the reservation with the times from result, and just mark it cancelled if it doesn't fit within opening times.

Reply:
I think in this case we should just use the times requested and create the reservation as DENIED

Reply:
Added the logic and a test for it.